python.html.markdown

```diff
@@ -995,32 +995,39 @@ values = (-x for x in [1,2,3,4,5])
 gen_to_list = list(values)
 print(gen_to_list)  # => [-1, -2, -3, -4, -5]
 
+# Function Annotations
+# Annotations for functions are completely optional
+# they are only for documentation purposes.
+# Return annotations are defined after `->` followed by an expression.
+def message(name: str, age: int) -> str:
+    msg = "{name} is {age} years old!".format(name=name, age=age)
+    # Return type is not enforced
+    return msg
 
 # Decorators
-# In this example `beg` wraps `say`. If say_please is True then it
-# will change the returned message.
-from functools import wraps
-
-
-def beg(target_function):
-    @wraps(target_function)
+# In this example `beg` wraps `say`. 
+# Beg extends the behavior of the function `say` without modifying it.
+# Using decorators you can change behavior of functions before and after.
```

Comment:
change behavior of functions by adding additional logic before and after the functions themselves